app/models/assignment.rb

```diff
@@ -37,13 +38,19 @@ class Assignment < ActiveRecord::Base
 
   scope :excluding_incomplete_features_for, ->(app_build) do
     joins(:split).where(
-      Arel::Nodes::Or.new(
-        arel_table[:force].eq(true),
-        Split.arel_excluding_incomplete_features_for(app_build)
+      Arel::Nodes::Grouping.new(
+        arel_table[:force].eq(true)
+        .or(Split.arel_excluding_incomplete_features_for(app_build))
       )
     )
   end
 
+  scope :excluding_remote_kills_for, ->(app_build) do
+    joins(:split).where(
+      Split.arel_excluding_remote_kills_for(app_build, override: arel_table[:force], overridden_at: arel_table[:updated_at])
```

Comment:
In order to make force assignments only win over remote kills when they're newer than remote kills, I ended up having to plumb some attributes through to the underlying scopes. In the process I learned some things that I now believe about the nuance of how force assignments should work with respect to feature completions and remote kills.

I now believe that a force assignment should always win over feature incompletion, even if a feature complete flag is set later. By definition you've already turned on the unfinished feature as an alpha tester or internal population. There's no reason that marking a flag feature complete should disable your access to the partially completed feature on your device. In practice, you're in a population that's likely to upgrade a lot and it may not matter much because you'll probably be on the feature complete version post-haste, but I'd argue that it'd be more surprising as an alpha tester or developer to see the feature shut itself down for you on your device as a result of marking a future version as feature complete.

Remote kills, on the other hand, are likely to be deployed in response to crash-bugs or similarly bad news incidents. If I'm a developer, I probably don't want my app to keep blowing up after a remote kill is put in place - I want my app to respect the remote kill immediately, but give me the option to toggle the feature back on if I want to see the faulty behavior for debugging purposes.

So this behavior offers us that. force assignments always win over feature incompleteness, but they only win over remote kills updated before our most recent assignment update.

Reply:
Strongly agreeing with paragraph two. Paragraph 3 seems reasonable.